homeassistant/components/google_generative_ai_conversation/conversation.py

```diff
@@ -226,9 +217,23 @@ async def async_process(
                     device_id=user_input.device_id,
                 )
 
-                prompt = (
-                    await llm_api.async_get_api_prompt(empty_tool_input) + "\n" + prompt
+                api_prompt = await llm_api.async_get_api_prompt(empty_tool_input)
+
+            else:
+                api_prompt = llm.PROMPT_NO_API_CONFIGURED
+
+            prompt = (
+                api_prompt
+                + "\n"
```

Comment:
maybe `"\n".join`?

Might not be any faster though since the python overhead probably exceeds the cost of making 2 strings